examples/triangle_intersection/triangle_intersection.cpp

```diff
@@ -0,0 +1,384 @@
+  ArborX::ExperimentalHyperGeometry::Point<2> beta;
+  ArborX::ExperimentalHyperGeometry::Point<2> p0;
+
+  // x = a + alpha * (b - a) + beta * (c - a)
+  //   = (1-beta-alpha) * a + alpha * b + beta * c
+  KOKKOS_FUNCTION
+  Mapping(ArborX::ExperimentalHyperGeometry::Triangle<2> const &triangle)
+  {
+    auto const &a = triangle.a;
+    auto const &b = triangle.b;
+    auto const &c = triangle.c;
+
+    ArborX::ExperimentalHyperGeometry::Point<2> u = {b[0] - a[0], b[1] - a[1]};
+    ArborX::ExperimentalHyperGeometry::Point<2> v = {c[0] - a[0], c[1] - a[1]};
+
+    float const inv_det = 1. / (v[1] * u[0] - v[0] * u[1]);
+
+    alpha = ArborX::ExperimentalHyperGeometry::Point<2>{v[1] * inv_det,
+                                                        -v[0] * inv_det};
+    beta = ArborX::ExperimentalHyperGeometry::Point<2>{-u[1] * inv_det,
+                                                       u[0] * inv_det};
+    p0 = a;
+  }
+
+  KOKKOS_FUNCTION ArborX::Point
+  get_coeff(ArborX::ExperimentalHyperGeometry::Point<2> p) const
+  {
+    float alpha_coeff = alpha[0] * (p[0] - p0[0]) + alpha[1] * (p[1] - p0[1]);
+    float beta_coeff = beta[0] * (p[0] - p0[0]) + beta[1] * (p[1] - p0[1]);
+    return {1 - alpha_coeff - beta_coeff, alpha_coeff, beta_coeff};
+  }
+
+#ifndef NDEBUG
+  // Recover the triangle from the mapping. Only used for debugging.
+  KOKKOS_FUNCTION ArborX::ExperimentalHyperGeometry::Triangle<2>
+  get_triangle() const
+  {
+    float const inv_det = 1. / (alpha[0] * beta[1] - alpha[1] * beta[0]);
+    ArborX::ExperimentalHyperGeometry::Point<2> a = p0;
+    ArborX::ExperimentalHyperGeometry::Point<2> b = {
+        {p0[0] + inv_det * beta[1], p0[1] - inv_det * beta[0]}};
+    ArborX::ExperimentalHyperGeometry::Point<2> c = {
+        {p0[0] - inv_det * alpha[1], p0[1] + inv_det * alpha[0]}};
+    return {a, b, c};
+  }
+#endif
+};
+
+// Store the points that represent the queries.
+template <typename DeviceType>
+class Points
+{
+public:
+  Points(typename DeviceType::execution_space const &execution_space)
+  {
+    initialize(execution_space);
+  }
+
+  void initialize(typename DeviceType::execution_space const &execution_space)
+  {
+    points_ = Kokkos::View<ArborX::ExperimentalHyperGeometry::Point<2> *,
+                           typename DeviceType::memory_space>(
+        Kokkos::view_alloc(Kokkos::WithoutInitializing, "points"), 2 * n);
+
+    Kokkos::parallel_for(
+        Kokkos::MDRangePolicy<Kokkos::Rank<2>,
+                              typename DeviceType::execution_space>(
+            execution_space, {0, 0}, {nx, ny}),
+        KOKKOS_CLASS_LAMBDA(int i, int j) {
+          auto index = [](int i, int j) { return i + j * nx; };
+
+          points_[2 * index(i, j)] = {(i + .25f) * hx, (j + .25f) * hy};
+          points_[2 * index(i, j) + 1] = {(i + .75f) * hx, (j + .75f) * hy};
+        });
+  }
+
+  KOKKOS_FUNCTION auto const &get_point(int i) const { return points_(i); }
```

Comment:
May want 
```suggestion
  KOKKOS_FUNCTION auto operator()(int i) const { return points_(i); }
```
for the small size 2D points.